dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/monitor/DatadogRumMonitor.kt

```diff
@@ -0,0 +1,124 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-2020 Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.monitor
+
+import com.datadog.android.core.internal.data.Writer
+import com.datadog.android.core.internal.time.TimeProvider
+import com.datadog.android.core.internal.utils.devLogger
+import com.datadog.android.rum.RumContext
+import com.datadog.android.rum.RumMonitor
+import com.datadog.android.rum.internal.domain.RumEvent
+import com.datadog.android.rum.internal.domain.RumEventData
+import java.util.UUID
+
+internal class DatadogRumMonitor(
+    applicationId: UUID,
+    private val timeProvider: TimeProvider,
+    private val writer: Writer<RumEvent>
+) : RumMonitor {
+
+    private var context: RumContext =
+        RumContext(
+            applicationId,
+            UUID.randomUUID(),
+            null
+        )
+
+    private val activeView = mutableMapOf<Any, RumEvent>()
```

Comment:
I this synchronised? If public methods are called from different threads, we may have read / write issue on this, no?

Reply:
Possibly, I'll add tests on that.